### PR TITLE
Add MRR (Mean Reciprocal Rank) metric to rec_sys

### LINKED ISSUE #3569: [Feature]: Add Mean Reciprocal Rank (MRR) metric to rec_sys
🚀 Feature

MRR (Mean Reciprocal Rank) is a widely used metric for evaluating retrieval systems, recommendation systems, and RAG pipelines. Currently Ignite has HitRate in rec_sys but MRR is missing.

Steps to reproduce :
Add a MRR metric to ignite/metrics/rec_sys/ similar to the existing HitRate implementation. MRR measures the average of 
1/rank of the first relevant item across all queries, which is particularly useful for evaluating retrieval quality in RAG systems.

I am interested in contributing this feature.

### PR DESCRIPTION
**Pull request by steaphenai**
closes #3569
Refs #3572

Description:  Add MRR (Mean Reciprocal Rank) metric to `ignite/metrics/rec_sys`.

- Supports multiple `top_k` values in a single call
- Uses `torch.argsort` with `stable=True` for deterministic tie-breaking
- Uses ranx as reference implementation for tests
- Adds `relevance_threshold` to support graded relevance labels

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/rec_sys/test_mrr_metric.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+
+def ranx_mrr(
+    y_pred: np.ndarray,
+    y: np.ndarray,
+    top_k: list[int],
+    ignore_zero_hits: bool = True,
+) -> list[float]:
+    """Reference MRR implementation using ranx for verification."""
+    from ranx import Qrels, Run, evaluate
```
Comment (vfdev-5) [inline tests/ignite/metrics/rec_sys/test_mrr_metric.py:18]: Put the link on the package in the description and we also need to add it into requirements-dev.txt

**Comment by vfdev-5** [comment]
@steaphenai please check also this discussion on top k stuff: https://github.com/pytorch/ignite/issues/3568#issuecomment-3939547890

**Comment by steaphenai** [comment]
> @steaphenai please check also this discussion on top k stuff: [#3568 (comment)](https://github.com/pytorch/ignite/issues/3568#issuecomment-3939547890)

Thanks! I've read #3568. The current implementation keeps top_k specific to MRR following the same pattern as HitRate. I'll update if needed once the TopK wrapper approach is decided.

**Comment by vfdev-5** [comment]
@aaishwarymishra can you review this PR instead of others and make it up to a mergeable state? Thanks!

**Comment by aaishwarymishra** [comment]
@vfdev-5 My bad, I missed that there already a pr open :)

File: ignite/metrics/rec_sys/mrr.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        device: str | torch.device = torch.device("cpu"),
+        skip_unrolling: bool = False,
+    ):
+        if any(k <= 0 for k in top_k):
+            raise ValueError(" top_k must be list of positive integers only.")
+
+        self.top_k = sorted(top_k)
+        self.ignore_zero_hits = ignore_zero_hits
+        self.relevance_threshold = relevance_threshold
+        super(MRR, self).__init__(output_transform, device=device, skip_unrolling=skip_unrolling)
```
Comment (aaishwarymishra) [inline ignite/metrics/rec_sys/mrr.py]:
```suggestion
        super().__init__(output_transform, device=device, skip_unrolling=skip_unrolling)
```

File: ignite/metrics/rec_sys/mrr.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        y_pred, y = output
+        if y_pred.shape != y.shape:
+            raise ValueError(f"y_pred and y must be in the same shape, got {y_pred.shape} != {y.shape}.")
+
+        if self.ignore_zero_hits:
+            valid_mask = torch.any(y >= self.relevance_threshold, dim=-1)
+            y_pred = y_pred[valid_mask]
+            y = y[valid_mask]
+
+        if y.shape[0] == 0:
```
Comment (aaishwarymishra) [inline ignite/metrics/rec_sys/mrr.py]: this shape check should be at top

File: ignite/metrics/rec_sys/mrr.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            return
+
+        max_k = self.top_k[-1]
+
+        # stable=True ensures deterministic tie-breaking, consistent with
+        # reference libraries such as ranx.
+        ranked_indices = torch.argsort(y_pred, dim=-1, descending=True, stable=True)[:, :max_k]
```
Comment (aaishwarymishra) [inline ignite/metrics/rec_sys/mrr.py]: we can simply use torch.topk

File: ignite/metrics/rec_sys/mrr.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            has_hit = relevant_mask.any(dim=-1)
+
+            # argmax on int tensor returns 0-based position of first True
+            first_hit_pos = relevant_mask.int().argmax(dim=-1)
+
+            reciprocal_rank = torch.where(
+                has_hit,
+                1.0 / (first_hit_pos.float() + 1.0),
+                torch.zeros_like(first_hit_pos, dtype=torch.float),
+            )
```
Comment (aaishwarymishra) [inline ignite/metrics/rec_sys/mrr.py:169]: I think the precision of the float should be same as the y_pred and y

File: ignite/metrics/rec_sys/mrr.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    def reset(self) -> None:
+        self._sum_reciprocal_ranks_per_k = torch.zeros(len(self.top_k), device=self._device)
+        self._num_examples = 0
+
+    @reinit__is_reduced
+    def update(self, output: tuple[torch.Tensor, torch.Tensor]) -> None:
+        if len(output) != 2:
+            raise ValueError(f"output should be in format `(y_pred,y)` but got tuple of {len(output)} tensors.")
+
+        y_pred, y = output
```
Comment (aaishwarymishra) [inline ignite/metrics/rec_sys/mrr.py]: we can detach tensors to be more efficient

File: ignite/metrics/rec_sys/mrr.py (hunk 3 above)
Comment (aaishwarymishra) [reply ignite/metrics/rec_sys/mrr.py]: My bad now I look at the code and notice it was correct last time maybe you can revert change for this code block :)
